### PR TITLE
Prometheus: Fix ad-hoc filters

### LINKED ISSUE #66520: Prometheus: Ad hoc filters not working in 9.5
**What happened**:
Looks like we renamed getTagKeys which is required by the parent datasource class if that datasource provides ad-hoc filters. Let's change it back!

**What you expected to happen**:
Ad-hoc filters work with prometheus datasources.

**How to reproduce it (as minimally and precisely as possible)**:
Try to use ad-hoc filters on 9.5

**Anything else we need to know?**:
Introduced in https://github.com/grafana/grafana/pull/63529/, will need backport into 9.5.x

**Environment**:
- Grafana version: 9.5
- Data source type & version:
- OS Grafana is installed on:
- User OS & Browser:
- Grafana plugins:
- Others:

### PR DESCRIPTION
**Pull request by gtk-grafana**
Refactor label names function back to getTagKeys to fulfill contract with parent class required for ad-hoc functionality

**What is this feature?**
Fix ad-hoc filter functionality with prometheus datasources in 9.5

Fixes https://github.com/grafana/grafana/issues/66520

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [x] If this is a pre-GA feature, it is behind a feature toggle.
- [x] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by joey-grafana — APPROVED** [review]

**Comment by gtk-grafana** [comment]
@bohandley I can verify that this fixes the ad-hoc issue, but wondering if you can take a quick look and see if I broke or missed anything!

File: public/app/plugins/datasource/prometheus/datasource.test.ts [hunk 1]
```diff
@@ -94,6 +94,16 @@ describe('PrometheusDatasource', () => {
     ds = new PrometheusDatasource(instanceSettings, templateSrvStub, timeSrvStub);
   });
 
+  // Some functions are required by the parent datasource class to provide functionality such as ad-hoc filters, which requires the definition of the getTagKeys, and getTagValues functions
```
Comment (itsmylife) [inline public/app/plugins/datasource/prometheus/datasource.test.ts:97]: That's a good comment. I was gonna remove them here https://github.com/grafana/grafana/pull/66466/files#diff-b6a6838d619292fa468c692a4a246b033da56d8dd5d4cad9d47a549345f59780L557-L568

Can you point me where exactly those are being called?

Comment (gtk-grafana) [reply public/app/plugins/datasource/prometheus/datasource.test.ts:97]: [AdHocFilterKey.tsx](https://github.com/grafana/grafana/blob/main/public/app/features/variables/adhoc/picker/AdHocFilterKey.tsx)
```
if (!ds || !ds.getTagKeys) {
    return [];
  }
```

**Review by itsmylife — APPROVED** [review]

**Comment by itsmylife** [comment]
@gtk-grafana I did not test locally. I checked the code only. FYI

**Review by bohandley — APPROVED** [review]
Nice work @gtk-grafana! Ad hoc filters work in prometheus with this, can confirm. I played around with some variables too and didn't see any funky behavior. 🚀 

https://user-images.githubusercontent.com/25674746/231869716-e6d0f5a7-4bac-438f-9840-60a820ef6872.mov

**Pull request merged**